File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-4.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        count = 0
        if not root:
            return count
        def traverse(node, val = None):
            nonlocal count
            if not val or node.val >= val:
                val = node.val
                count +=1
            if node.left:
                traverse(node.left, val)
            if node.right:
                traverse(node.right, val)
        traverse(root)
        return count 

File: Data Structures & Algorithms/invert-a-binary-tree/submission-4.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    private void swap(TreeNode node) {
        if (node!= null) {
            TreeNode left = node.left;
            node.left = node.right;
            node.right = left;
            swap(node.left);
            swap(node.right);
        }
    }
    public TreeNode invertTree(TreeNode root) {
        swap(root);
        return root;
        
    }
}
